apps/coscrad-frontend/src/components/higher-order-components/categorizable-page-layout.tsx

```diff
@@ -0,0 +1,132 @@
+import { AggregateCompositeIdentifier } from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    Close as CloseIcon,
+    Hub as HubIcon,
+    TextSnippet as TextSnippetIcon,
+} from '@mui/icons-material';
+import { Box, Drawer, IconButton, Stack, Tooltip, Typography, styled } from '@mui/material';
+import { ReactNode, useState } from 'react';
+import { CommandPanel } from '../commands';
+
+interface CategorizablePageLayoutProps {
+    compositeIdentifier: AggregateCompositeIdentifier;
+    selfNotesList: JSX.Element;
+    connectedResourcesList: JSX.Element;
+    commandPanel?: JSX.Element;
+    children: ReactNode;
+}
+
+const DrawerHeader = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    alignItems: 'flex-start',
+    margin: theme.spacing(2, 4, 0, 4),
+    justifyContent: 'space-between',
+}));
+
+const DrawerContentStack = styled(Stack)(({ theme }) => ({
+    margin: theme.spacing(0, 0, 0, 4),
+}));
+
+export const CategorizablePageLayout = ({
+    compositeIdentifier: { type: resourceType, id },
+    selfNotesList,
+    connectedResourcesList,
+    commandPanel,
+    children,
+}: CategorizablePageLayoutProps) => {
+    const [bottomDrawerOpen, setBottomDrawerOpen] = useState(false);
```

Comment:
If you keep the current design, you should name the boolean flags according to convention.
`isBottomDrawerOpen`, `setIsBottomDrawerOpen`, and similarly for the one below.